timeseries/src/autogluon/timeseries/models/local/statsforecast.py

```diff
@@ -0,0 +1,252 @@
+import logging
+from typing import List, Type, Union
+
+import pandas as pd
+
+from autogluon.core.utils.exceptions import TimeLimitExceeded
+from autogluon.timeseries.dataset.ts_dataframe import ITEMID, TIMESTAMP, TimeSeriesDataFrame
+from autogluon.timeseries.models.local.abstract_local_model import AbstractLocalModel
+from autogluon.timeseries.utils.hashing import hash_ts_dataframe_items
+
+logger = logging.getLogger(__name__)
+
+
+class AbstractStatsForecastModel(AbstractLocalModel):
+    """Wrapper for StatsForecast models.
+
+    Cached predictions are stored inside the model to speed up validation & ensemble training downstream.
+
+    Attributes
+    ----------
+    allowed_local_model_args : List[str]
+        List of allowed arguments that can be passed to the underlying model.
+        Arguments not in this list will be filtered out and not passed to the underlying model.
+    """
+
+    allowed_local_model_args: List[str] = []
+    DEFAULT_N_JOBS: Union[float, int] = -1
+
+    def get_model_type(self) -> Type:
+        raise NotImplementedError
+
+    def _fit(self, train_data, time_limit=None, verbosity=2, **kwargs) -> None:
+        """Prepare hyperparameters that will be passed to the underlying model.
+
+        As for all local models, actual fitting + predictions are delegated to the ``predict`` method.
+        """
+        # TODO: Find a way to ensure that SF models respect time_limit
+        # Fitting usually takes >= 20 seconds
+        if time_limit is not None and time_limit < 20:
```

Comment:
The `fit` time should be dependent on instance type as well. If `time_limit` is not implemented and we expect the local model fit shall be quick compared with other models, shall we send warning message instead?

Reply:
Updates this logic based on the discussion below.